pybaseball/playerid_lookup.py

```diff
@@ -108,19 +104,16 @@ def search(self, last: str, first: str = None, fuzzy: bool = False) -> pd.DataFr
         if len(results) == 0 and fuzzy:
             print("No identically matched names found! Returning the 5 most similar names.")
             similar_names_df = name_similarity(last=last, first=first, player_table=self.table)
-            results = similar_names_df.merge(self.table, left_on=["name_last", "name_first"], right_on=[
-                                             "name_last", "name_first"], how="left")
+            results = self.table.iloc[similar_names_df].reset_index(drop=True)
```

Comment:
we have `name_similarity` returning a list of indexes, and then we subset with `iloc` and return that. It seems like we don;t need two steps here, `name_similarity` could do the subsetting and return the result dataframe?

i.e., 
```
return name_similarity(last=last, first=first, player_table=self.table)
```

Reply:
Done